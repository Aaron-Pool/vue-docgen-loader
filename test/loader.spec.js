import compiler from './compiler'

it('Injects docgen result as component.options.__docgenInfo', async () => {
  const stats = await compiler('./fixtures/basic-sfc.vue')
  const output = stats.toJson().modules[0].modules[0].source

  const docgenPattern = /component\.options\.__docgenInfo\s?=\s?(\{[\s\S]*})/

  expect(output).toMatch(docgenPattern)
  expect(JSON.parse(output.match(docgenPattern)[1])).toMatchSnapshot()
})
