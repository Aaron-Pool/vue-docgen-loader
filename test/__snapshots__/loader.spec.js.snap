// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Injects docgen result as component.options.__docgenInfo 1`] = `
Object {
  "description": "",
  "displayName": "basic-sfc",
  "events": Object {},
  "methods": Array [],
  "props": Object {
    "foo": Object {
      "defaultValue": Object {
        "func": false,
        "value": "'bar'",
      },
      "description": "Description of foo.",
      "name": "foo",
      "required": "",
      "tags": Object {},
      "type": Object {
        "name": "string",
      },
    },
  },
  "slots": Object {},
  "tags": Object {},
}
`;
